tests/lax_test.py

```diff
@@ -2245,6 +2245,25 @@ def test_select_jvp_complexity(self):
                                              (x,), (1.,)))(1.)
     self.assertLen(jaxpr.jaxpr.eqns, 2)
 
+  def test_named_call(self):
+
+    def named_call(f, name):
```

Comment:
I think this `named_call` should be a public function in `lax`. We do not want users having to bind the primitive themselves. I'm happy for that to be done in a follow up PR though since having a single shared primitive is already a strict improvement.

Reply:
Thanks for the review! A named_call public function is great. Let me think about how to implement it since it is where the stateful_named_call in haiku and named_call in flax diverges. Let me start a draft for feedback first.

Reply:
Please feel free to just make this a TODO for now.

Reply:
Done.